server/session/handler_block_actor_data.go

```diff
@@ -0,0 +1,29 @@
+package session
+
+import (
+	"fmt"
+	"github.com/df-mc/dragonfly/server/block"
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/sandertv/gophertunnel/minecraft/protocol/packet"
+)
+
+type BlockActorDataHandler struct {}
+
+func (b BlockActorDataHandler) Handle(p packet.Packet, s *Session) error {
+	pk := p.(*packet.BlockActorData)
+	x, y, z := int(pk.Position.X()), int(pk.Position.Y()), int(pk.Position.Z())
+	switch pk.NBTData["id"]{
+
+	case "Sign":
+		blockPosition := cube.Pos{x, y, z}
+		if v, ok := s.c.World().Block(blockPosition).(block.WoodSign); ok {
+			v.SignTextColor = pk.NBTData["SignTextColor"].(int32)
+			v.Text = pk.NBTData["Text"].(string)
```

Comment:
Make sure to validate this field and make sure the size is valid.
I think PocketMine checks if the Text is less than 1000, and won't update the block if it is over that threshold.